admin-api-server/src/nft/service/file/file.service.spec.ts

```diff
@@ -0,0 +1,357 @@
+    file.fieldname === 'files[]' &&
+    Buffer.isBuffer(file.buffer) &&
+    file.encoding === '7bit' &&
+    (file.originalname === 'display' || file.originalname === 'thumbnail')
+  );
+};
+
+const expectedDimensions = (file: File, width: number, height: number) => {
+  const dimensions = sizeOf(file.buffer);
+  return dimensions.width === width && dimensions.height === height;
+};
+
+describe('FileService', () => {
+  let fileService: FileService;
+
+  beforeEach(() => {
+    fileService = new FileService();
+  });
+
+  describe('#addMissingFiles should return artifact, thumbnail and display', () => {
+    describe('if the artifact is a video', () => {
+      describe('when display and thumbnail are missing', () => {
+        it('display should be created with size 350x350 and thumbnail with size 250x250', async () => {
+          const buffer = fs.readFileSync(__dirname + '/shortvideo.spec.mov');
+          const artifact = artifactMock(buffer, 'video');
+          const filesArray = [artifact];
+
+          const res = await fileService.addMissingFiles(filesArray);
+          const display = res.find((file) => file.originalname === 'display');
+          const thumbnail = res.find(
+            (file) => file.originalname === 'thumbnail',
+          );
+
+          expect(res.length).toBe(3);
+          expect(isImage(display!)).toBe(true);
+          expect(isImage(thumbnail!)).toBe(true);
+
+          expect(expectedDimensions(display!, 350, 350)).toBe(true);
+          expect(expectedDimensions(thumbnail!, 250, 250)).toBe(true);
+
+          const artifactAfterManipulation = res.find(
+            (file) => file.originalname === 'artifact',
+          );
+          expect(artifactAfterManipulation).toStrictEqual(artifact);
+        });
+      });
+
+      describe('when only thumbnail is missing, thumbnail should be created from display', () => {
+        it('with same size, when the display size is smaller than 350', async () => {
+          const buffer = fs.readFileSync(__dirname + '/shortvideo.spec.mov');
+          const artifact = artifactMock(buffer, 'video');
+          const display = await displayMock(300, 320);
+          const filesArray = [artifact, display];
+
+          const res = await fileService.addMissingFiles(filesArray);
+          const thumbnail = res.find(
+            (file) => file.originalname === 'thumbnail',
+          );
+
+          expect(res.length).toBe(3);
+          expect(isImage(thumbnail!)).toBe(true);
+
+          expect(expectedDimensions(thumbnail!, 300, 320)).toBe(true);
+
+          const afterManipulation = res.filter(
+            (file) =>
+              file.originalname === 'artifact' ||
+              file.originalname === 'display',
+          );
+          expect(afterManipulation).toStrictEqual([artifact, display]);
+        });
+
+        it('scaled down to size 250, when the display size is larger then 350', async () => {
```

Comment:
I think this doesn't make so much sense. We should simply be doing:
- if display larger than 350x350 (in whichever dimension), create scaled down thumbnail (respecting the same aspect ratio, as is already done if I see correctly)
- else, if display is 350x350 or smaller, thumbnail should equal display (no transformation)

And, thumbnail should target in either width or height 350 pixels. Not 250.

Reply:
Should be correct now.